src/process-events.c

```diff
@@ -374,40 +349,48 @@ static __always_inline int process_argv(struct pt_regs *ctx,
                                         u32 tail_index) {
     u64 pid_tgid = bpf_get_current_pid_tgid();
     u64 *idp = bpf_map_lookup_elem(&process_ids, &pid_tgid);
-    if (idp == NULL)
-    {
-        return 0;
-    }
-    u64 id = *idp;
+    if (idp == NULL) return 0;
 
-    // put the event on the stack and satisfy the verifier
-    telemetry_event_t tev = {0};
-    ptelemetry_event_t ev = &tev;
+    telemetry_event_t tev = {
+        .id = *idp,
+        .telemetry_type = TE_COMMAND_LINE,
+        .u.v = {
+            .value[0] = '\0',
+            .truncated = FALSE,
+        },
+    };
 
-    u32 index = 0;
-    u32 *pii = bpf_map_lookup_elem(&read_flush_index, &index);
-    if (NULL == pii)
-    {
-        bpf_map_update_elem(&read_flush_index, &index, &index, BPF_ANY);
-        goto Tail;
-    }
+    u32 always_zero = 0;
+    u32 arg_num = 0;
 
-    u32 ii = *pii;
-    char *ptr = NULL;
-    int ret = 0;
+    // check if we have tailed-back before; if so set arg_num
+    u32 *parg_num = bpf_map_lookup_elem(&read_flush_index, &always_zero);
+    if (parg_num != NULL) {
+        arg_num = *parg_num;
+    }
 
     // this number was arrived at experimentally, increasing it will result in too many
     // instructions for older kernels
-    READ_LOOP_N(argv, TE_COMMAND_LINE, 6);
-
-    bpf_map_update_elem(&read_flush_index, &index, &ii, BPF_ANY);
+    #pragma unroll
+    for (int i = 0; i < 6; i++) {
```

Comment:
Not for this PR, but it may be worth going back and seeing if we can raise some of these limits since `#pragma unroll` should generate fewer instructions.

Reply:
Yeah I considered it, especially for the `argv` program where we now have a lot of head room but decided to only try that after I add the code to address the race condition since that was going to add instructions already.